### ISSUE TITLE
ValueError: The configured object store size exceeds /dev/shm size

### ISSUE THREAD

**Issue by MortenKoppel** [alumae/kiirkirjutaja#5]
On bare metal server the ray app inside the container throws the following memory calculation? error and halts 
(system Ubuntu 22.04 LTS  5.15.0-30-generic and Docker version 20.10.16, build aa7e414, the kiirkirjutaja container v 0.2.2 and 0.2.7) :
~~~~
2022-05-16 20:36:49,653 INFO services.py:1267 -- View the Ray dashboard at http://127.0.0.1:8265/
Traceback (most recent call last):
File "main.py", line 34, in <module>
ray.init(num_cpus=4)
File "/root/miniconda3/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 47, in wrapper
return func(*args, **kwargs)
File "/root/miniconda3/lib/python3.8/site-packages/ray/worker.py", line 699, in init
_global_node = ray.node.Node(
File "/root/miniconda3/lib/python3.8/site-packages/ray/node.py", line 234, in __init__
self.start_ray_processes()
File "/root/miniconda3/lib/python3.8/site-packages/ray/node.py", line 894, in start_ray_processes
ray._private.services.determine_plasma_store_config(
File "/root/miniconda3/lib/python3.8/site-packages/ray/_private/services.py", line 1706, in determine_plasma_store_config
raise ValueError(
ValueError: The configured object store size (11.81114368 GB) exceeds /dev/shm size (11.8111232 GB). This will harm performance. Consider deleting files in /dev/shm or increasing its size with --shm-size in Docker. To ignore this warning, set RAY_OBJECT_STORE_ALLOW_SLOW_STORAGE=1.
~~~~
On other systems the container has worked without errors. Tried to debug without success. Seems to have something to do with object store size calculation in ray (one similar issue https://github.com/ray-project/ray/issues/14535).  Additional output with defined shm:
~~~~
1. Run container with dedicated shm size
root@kiirkirjutaja:~# docker run --shm-size=11000000000b ....
2. Check the shm size inside the container
Filesystem 1B-blocks Used Available Use% Mounted on
shm 11000000512 0 11000000512 0% /dev/shm
3. Try to run main.py from container, and receive the error
ValueError: The configured object store size (10.999984128 GB) exceeds /dev/shm size (10.999963648 GB).
4. Check shm size again from inside the container
Filesystem 1B-blocks Used Available Use% Mounted on
shm 11000000512 32768 10999967744 1% /dev/shm
~~~~

**Comment by MortenKoppel** [comment]
The only workaround at the moment is to modify main.py inside the running container and manually enter Object Store Memory Size to ray.init() function, for example 
`ray.init(object_store_memory=20000000000, num_cpus=4). `
Keeping in mind that object_store_memory<=shm. And save the running container as new image with
`docker commit <running_container_name> <new_image_name>`
The test machine has ~160 GB RAM (164711904), maybe it is somehow causing Ray to miscalculate memory.

**Comment by MortenKoppel** [comment]
Issue seems to be resolved in version 0.3.0.
